driver/kubernetes/kubernetes.go

```diff
@@ -291,34 +293,54 @@ func (k *Driver) streamPodLogs(options metav1.ListOptions, out io.Writer, done c
 	go func() {
 		// Track pods whose logs have been streamed by pod name. We need to know when we've already
 		// processed logs for a given pod, since multiple lifecycle events are received per pod.
-		streamedLogs := map[string]bool{}
+		streamedLogs := map[string]int{}
 		for event := range watcher.ResultChan() {
 			pod, ok := event.Object.(*v1.Pod)
 			if !ok {
 				continue
 			}
 			podName := pod.GetName()
-			if streamedLogs[podName] {
-				// The event was for a pod whose logs have already been streamed, so do nothing.
+
+			// The event was for a pod whose logs have already been streamed, so do nothing.
+			if streamedLogs[podName] >= numBackoffLoops {
 				continue
 			}
-			req := k.pods.GetLogs(podName, &v1.PodLogOptions{
-				Container: k8sContainerName,
-				Follow:    true,
-			})
-			reader, err := req.Stream()
-			// There was an error connecting to the pod, so continue the loop and attempt streaming
-			// logs again next time there is an event for the same pod.
-			if err != nil {
-				continue
+
+			timeout := 0
+			for (streamedLogs[podName] < numBackoffLoops) && (timeout < numBackoffLoops) {
```

Comment:
Unfortunately, the additional logic is becoming more convoluted rather than simpler.